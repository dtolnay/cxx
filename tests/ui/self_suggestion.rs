#[cxx::bridge]
mod ffi {
    extern "Rust" {
        fn self_rust(self);
    }

    extern "C++" {
        fn self_c(self);
    }
}

fn main() {}
