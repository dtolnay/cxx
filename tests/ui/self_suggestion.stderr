error: cannot taking self by-value, consider using `&self` or `&mut self` instead
 --> $DIR/self_suggestion.rs:4:22
  |
4 |         fn self_rust(self);
  |                      ^^^^

error: cannot taking self by-value, consider using `self: Pin<&mut Self>`, `self: Box<T>` or `self: UniquePtr<T>` instead
 --> $DIR/self_suggestion.rs:8:19
  |
8 |         fn self_c(self);
  |                   ^^^^
